care/emr/api/viewsets/report/report_upload.py

```
@@ -148,6 +152,59 @@ def generate(self, request, *args, **kwargs):
+    @extend_schema(
+        request=GenerateReportRequest, responses={200: "Success"}, tags=["report"]
+    )
+    @action(detail=False, methods=["POST"])
+    def preview(self, request, *args, **kwargs):
+        request_data = GenerateReportRequest.model_validate(request.data)
+        template = get_object_or_404(Template, external_id=request_data.template_id)
+
+        if not AuthorizationController.call(
+            "can_preview_report_from_template", request.user, template.facility
+        ):
+            raise PermissionDenied("You do not have permission to preview reports")
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)